src/averageAges.js

```diff
@@ -56,6 +61,50 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const filteredChildren = (!onlyWithSon)
+    ? people.filter(person => people
+      .find(mother => mother.name === person.mother))
+    : peopleFilter(
+      people.filter(person => people
+        .find(mother => mother.name === person.mother)),
+      'm'
+    );
+
+  const ageDifference = filteredChildren
+    .reduce((prev, curr) =>
+      (prev + curr.born - people
+        .find(person =>
+          person.name === curr.mother).born), 0);
+
+  return ageDifference / filteredChildren.length;
+}
+
+function peopleFilter(people, sex, century, withChildren) {
+  let filtered = people.filter(person => person.sex === sex);
+
+  if (century) {
+    filtered = filtered.filter(person =>
+      Math.ceil(person.died / 100) === century
+    );
+  }
+
+  if (withChildren) {
+    filtered = filtered.filter(person => people.find(
+      child => child.mother === person.name
+    ));
+  }
+
+  return filtered;
+}
```

Comment:
Again, here you've used filter method twice, but you can do it using it only once, just write needed condition inside it